src/test/mocks/archiveLoader.ts

```diff
@@ -83,11 +84,17 @@ export default () => {
     },
     mockPage: (parentBook: ArchiveBook, newPage: ArchivePage, pageSlug: string) => {
       localBookPages[`${parentBook.id}@${parentBook.version}`][newPage.id] = newPage;
-      localBooks[`${parentBook.id}@${parentBook.version}`].tree.contents.push({
-        id: `${newPage.id}@${newPage.version}`,
-        slug: pageSlug,
-        title: newPage.title,
-      });
+      const currentBook = localBooks[`${parentBook.id}@${parentBook.version}`];
+      const treeNode = findArchiveTreeNodeById(parentBook.tree, newPage.id);
+      if (treeNode) {
+        currentBook.tree.contents.push(treeNode);
```

Comment:
if you've found the treenode in the tree already, you don' t have to push it on.

i guess we're assuming that the `parentBook` is already in `localBooks`, and not some other object with the same id and version, but idk what to do about that.